### PR TITLE
Use `enumerate` where applicable

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**

### CODE REVIEW

**Comment by FrancescAlted** [comment]
Any thoughts on my suggestions?

**Comment by DimitriPapadopoulos** [comment]
Which suggestions?

**Comment by FrancescAlted** [comment]
Can't you see comments like?

image

**Comment by DimitriPapadopoulos** [comment]
I have taken your comment into account, by changing `enumerate` to `zip`.

File: blosc2/core.py [hunk 1]
```diff
@@ -1275,23 +1275,23 @@ def compute_chunks_blocks(
             blocks = [blocks]
         if len(blocks) != len(shape):
             raise ValueError("blocks should have the same length than shape")
-        for i in range(len(blocks)):
-            if blocks[i] == 0:
+        for b, s in zip(blocks, shape):
```
Comment (FrancescAlted) [inline blosc2/core.py:1278]: Please use significant varnames

File: blosc2/core.py [hunk 2]
```diff
@@ -1275,23 +1275,23 @@ def compute_chunks_blocks(
...
+            if b == 0:
                 raise ValueError("blocks cannot contain 0 dimension")
-            if shape[i] == 1 and blocks[i] > shape[i]:
+            if s == 1 and b > s:
```
Comment (FrancescAlted) [inline blosc2/core.py:1281]: Ditto

File: blosc2/core.py [hunk 3]
```diff
@@ -1275,23 +1275,23 @@ def compute_chunks_blocks(
...
                 raise ValueError("blocks cannot be greater than shape if it is 1")
     if chunks:
         if not isinstance(chunks, tuple | list):
             chunks = [chunks]
         if len(chunks) != len(shape):
             raise ValueError("chunks should have the same length than shape")
-        for i in range(len(chunks)):
-            if shape[i] == 1 and chunks[i] > shape[i]:
+        for c, s in zip(chunks, shape):
```
Comment (FrancescAlted) [inline blosc2/core.py:1288]: ditto

File: blosc2/core.py [hunk 4]
```diff
@@ -1275,23 +1275,23 @@ def compute_chunks_blocks(
...
+            if s == 1 and c > s:
                 raise ValueError("chunks cannot be greater than shape if it is 1")
 
     if chunks is not None and blocks is not None:
-        for i in range(len(blocks)):
-            if blocks[i] > chunks[i]:
+        for b, c in zip(blocks, chunks):
```
Comment (FrancescAlted) [inline blosc2/core.py:1293]: ditto

File: tests/ndarray/test_auto_parts.py [hunk 5]
```diff
@@ -43,9 +43,9 @@ def test_compute_chunks_blocks(clevel, codec, shape: tuple, dtype):
     else:
         chunks, blocks = blosc2.compute_chunks_blocks(shape, **cparams)
     # print(chunks, blocks)
-    for i in range(len(shape)):
-        assert shape[i] >= chunks[i]
-        assert chunks[i] >= blocks[i]
+    for s, c, b in zip(shape, chunks, blocks):
```
Comment (FrancescAlted) [inline tests/ndarray/test_auto_parts.py:46]: ditto

File: tests/ndarray/test_auto_parts.py [hunk 6]
```diff
@@ -65,9 +65,9 @@ def test_compute_chunks_blocks(clevel, codec, shape: tuple, dtype):
 def test_compute_chunks(shape: tuple, blocks: tuple):
     chunks, blocks = blosc2.compute_chunks_blocks(shape, blocks=blocks)
     # print(chunks, blocks)
-    for i in range(len(shape)):
-        assert shape[i] >= chunks[i]
-        assert chunks[i] >= blocks[i]
+    for s, c, b in zip(shape, chunks, blocks):
```
Comment (FrancescAlted) [inline tests/ndarray/test_auto_parts.py:68]: ditto

**Comment by FrancescAlted** [comment]
> I can't see the comments. Perhaps you haven't submitted the review — hence the “Pending” icon.

Ah, that was it.  Sorry about that.

File: blosc2/core.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply blosc2/core.py:1278]: I was unable to find good names. Perhaps `array_len`, `block_len`, `chunk_len`?

Comment (FrancescAlted) [reply blosc2/core.py:1278]: What about adopting the singular case?  e.g. `dim`,  `block`, `chunk` ?

Comment (DimitriPapadopoulos) [reply blosc2/core.py]: I have applied your suggestion. Alternatives:
* `dim`, `block_dim`, `chunk_dim`
* `array_dim`, `block_dim`, `chunk_dim`

File: blosc2/core.py [hunk 7]
```diff
@@ -1275,23 +1275,23 @@ def compute_chunks_blocks(
...
-            if shape[i] == 1 and chunks[i] > shape[i]:
+        for chunk, dim in zip(chunks, shape):
+            if dim == 1 and chunk > dim:
                 raise ValueError("chunks cannot be greater than shape if it is 1")
 
     if chunks is not None and blocks is not None:
-        for i in range(len(blocks)):
-            if blocks[i] > chunks[i]:
+        for block, chunck in zip(blocks, chunks):
+            if block > chunck:
```
Comment (FrancescAlted) [inline blosc2/core.py:1294]: Typo: chunck -> chunk

The same one happens below

**Review by FrancescAlted — APPROVED** [review]
LGTM

**Comment by FrancescAlted** [comment]
Thanks @DimitriPapadopoulos

**Pull request merged**